Setup.hs

```diff
@@ -66,12 +67,22 @@ copyHook' pd lbi hooks flags = do
     }
 
 -- Used to add .agdai files to data-files
-expandAgdaExt :: FilePath -> [FilePath]
-expandAgdaExt fp | takeExtension fp == ".agda" = [ fp, toIFile fp ]
-                 | otherwise                   = [ fp ]
+expandAgdaExt :: PackageDescription -> FilePath -> [FilePath]
+expandAgdaExt pd fp | takeExtension fp == ".agda" = [ fp, toIFile pd fp ]
+                    | otherwise                   = [ fp ]
 
-toIFile :: FilePath -> FilePath
-toIFile file = replaceExtension file ".agdai"
+version :: PackageDescription -> String
+version = intercalate "." . map show . versionNumbers . pkgVersion . package
+
+projectRoot :: PackageDescription -> FilePath
+projectRoot pd = takeDirectory agdaLibFile where
+  [agdaLibFile] = filter ((".agda-lib" ==) . takeExtension) $ dataFiles pd
+
+toIFile :: PackageDescription -> FilePath -> FilePath
+toIFile pd file = buildDir </> fileName where
+  root = projectRoot pd
+  buildDir = root </> "_build" </> version pd </> "agda"
+  fileName = makeRelative root $ replaceExtension file ".agdai"
```

Comment:
This code malfunctions.
We have (for some suitable `BASE`)
```
root = lib/prim
file = /BASE/src/data/lib/prim/Agda/Primitive.agda
fileName = /BASE/src/data/lib/prim/Agda/Primitive.agdai
   -- because makeRelative root silently does nothing when the root does not match
toIFile pd file =  /BASE/src/data/lib/prim/Agda/Primitive.agdai
   -- because `p </> a` is just `a` if `a is an absolute path
```